src/Avalonia.Base/Utilities/MathUtilities.cs

```diff
@@ -188,6 +188,11 @@ public static bool IsZero(float value)
         /// <returns>The clamped value.</returns>
         public static double Clamp(double val, double min, double max)
         {
+            if (min > max)
+            {
+                throw new ArgumentException($"{min} cannot be greater than {max}.");
```

Comment:
Perhaps we need a helper function here to help with inlining (like inhttps://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Math.cs#L340).